unit-tests/game-api-tests.js

```diff
@@ -1,32 +1,23 @@
-//describe('gameApiTest', function(){
-//
-//    var gameApi;
-//
-//    beforeEach(function(){
-//        module('tombola.noughtAndCrosses.services');
-//        inject(function($injector){
-//            gameApi = $injector.get('gameApi');
-//        });
-//    });
-//
-//    it('Should contain gameApi service', function(){
-//        expect(noughtsAndCrossesApp.gameApi)
-//            .to.equal('noughtsAndCrossesApp.service(gameApi,function ($http,$q,gameApiConstants)')
-//    });
-//
-//    it('makeMove function contains appropriate playerNumber and squareNumber', function(){
-//        expect(noughtsAndCrossesApp.gameApi.makeMove)
-//            .to.equal(gameApi.makeMove.playerNumber = 'playerNumber',gameApi.makeMove.squareNumber = 'squareNumber');
-//    });
-//
-//    it('newGame function contains constants', function(){
-//        expect(noughtsAndCrossesApp.gameApi.newGame)
-//            .to.equal(gameApi.newGame.player1 = 'player1Type',gameApi.newGame.player2 = 'player2Type');
-//    });
-//
-//    it('gameApi service contains promise', function(){
-//        expect(noughtsAndCrossesApp.gameApi)
-//            .to.equal(gameApi.callServer);
-//    });
-//
-//});
+describe('gameApiTest', function(){
+
+    var gameApi;
+    var gameModel;
+
+    beforeEach(function(){
```

Comment:
We don't use the mock $http object - the unit tests will call the actual service. This is bad as it means you need access to the server, for it to be up and to be able to predict it's response (e.g. what happens if you test random vs random). None of these can be relied up - mock the API service and provide it with hard-coded responses:
https://docs.angularjs.org/api/ngMock/service/$httpBackend
You should also test the proxy is sending the correct stuff up to the API as well by using the backend mock